src/io/flutter/console/FlutterConsoles.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2016 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.console;
+
+import com.intellij.execution.process.OSProcessHandler;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Key;
+import com.intellij.ui.content.Content;
+import com.intellij.ui.content.MessageView;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Methods that use the appropriate Flutter console.
+ * <p>
+ * We share one Flutter console per Module, and have one global Flutter console (for tasks which don't
```

Comment:
Not for this PR, but we may want to consider moving to having just one Flutter console per project (a bit simpler for the user than seeing multiple flutter console open at one time). I think this has the drawback that we won't be able to do as good a job at hyperlinking some content.